File: common/src/main/java/com/jmsoftware/maf/common/domain/authcenter/permission/GetPermissionListByRoleIdListResponse.java
package com.jmsoftware.maf.common.domain.authcenter.permission;

import lombok.Data;

import java.util.LinkedList;
import java.util.List;

/**
 * <h1>GetPermissionListByRoleIdListResponse</h1>
 * <p>
 * Change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com
 * @date 5 /11/20 8:38 AM
 */
@Data
public class GetPermissionListByRoleIdListResponse {
    /**
     * The Permission list.
     */
    private List<Permission> permissionList;

    @Data
    public static class Permission {
        /**
         * The Url.
         */
        private String url;
        /**
         * The Type.
         */
        private Byte type;
        /**
         * The Permission expression.
         */
        private String permissionExpression;
        /**
         * The Method.
         */
        private String method;
    }
}

File: common/src/main/java/com/jmsoftware/maf/common/domain/authcenter/permission/GetPermissionListByRoleIdListResponse.kt
package com.jmsoftware.maf.common.domain.authcenter.permission

/**
 * # GetPermissionListByRoleIdListResponse
 *
 * Change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/16/22 6:43 PM
 */
class GetPermissionListByRoleIdListResponse {
    /**
     * The Permission list.
     */
    var permissionList: MutableList<Permission> = mutableListOf()

    class Permission {
        /**
         * The Url.
         */
        var url: String? = null

        /**
         * The Type.
         */
        var type: Byte? = null

        /**
         * The Permission expression.
         */
        var permissionExpression: String? = null

        /**
         * The Method.
         */
        var method: String? = null
    }
}
